.github/workflows/check-catalyst.yaml

```diff
@@ -372,6 +373,90 @@ jobs:
         MDD_BENCHMARK_PRECISION=1 \
         pytest demos --nbmake -n auto
 
+  frontend-tests-lightning-kokkos:
+    name: Frontend Tests (backend="lightning.kokkos")
+    needs: [runtime, mhlo, quantum]
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Cancel previous runs
+      uses: styfle/cancel-workflow-action@0.11.0
+      with:
+        access_token: ${{ github.token }}
+
+    - name: Checkout Catalyst repo
+      uses: actions/checkout@v3
+
+    - name: Install Deps
+      run: |
+        sudo apt-get install -y python3 python3-pip libomp-dev
+        python3 -m pip install -r requirements.txt
+        python3 -m pip install .
+
+    - name: Get LLVM Version
+      id: llvm-hash
+      run: echo "llvm-hash=$(grep llvm .dep-versions | awk -F '=' '{ print $2 }')" >> $GITHUB_OUTPUT
+
+    - name: Get Cached LLVM Build
+      id: cache-llvm-build
+      uses: actions/cache@v3
+      with:
+        path: llvm-build
+        key: ${{ runner.os }}-llvm-${{ steps.llvm-hash.outputs.llvm-hash }}-default-build
+        fail-on-cache-miss: True
+
+    - name: Get MHLO Version
+      id: mhlo-hash
+      run: echo "mhlo-hash=$(grep mhlo .dep-versions | awk -F '=' '{ print $2 }')" >> $GITHUB_OUTPUT
+
+    - name: Get Cached MHLO Build
+      id: cache-mhlo
+      uses: actions/cache@v3
+      with:
+        path: mhlo-build
+        key: ${{ runner.os }}-mhlo-${{ steps.mhlo-hash.outputs.mhlo-hash }}-default-build
+        fail-on-cache-miss: True
+
+    - name: Download Quantum Build Artifact
+      uses: actions/download-artifact@v3
+      with:
+        name: quantum-build
+        path: quantum-build
+
+    - name: Download Catalyst-Runtime Artifact
+      uses: actions/download-artifact@v3
+      with:
+        name: runtime-build
+        path: runtime-build/lib
+
+    - name: Add Frontend Dependencies to PATH
+      run: |
+        echo "$(pwd)/llvm-build/bin" >> $GITHUB_PATH
+        echo "$(pwd)/mhlo-build/bin" >> $GITHUB_PATH
+        echo "$(pwd)/quantum-build/bin" >> $GITHUB_PATH
+        echo "PYTHONPATH=$PYTHONPATH:$(pwd)/quantum-build/python_packages/quantum" >> $GITHUB_ENV
+        echo "RUNTIME_LIB_DIR=$(pwd)/runtime-build/lib" >> $GITHUB_ENV
+        echo "MLIR_LIB_DIR=$(pwd)/llvm-build/lib" >> $GITHUB_ENV
+        chmod +x quantum-build/bin/quantum-opt  # artifact upload does not preserve permissions
+
+    - name: Checkout Lightning-Kokkos repository
+      uses: actions/checkout@v3
+      with:
+        repository: PennyLaneAI/pennylane-lightning-kokkos
+        ref: v0.30.0
+        path: pennylane-lightning-kokkos
+
+    - name: Install lightning.kokkos used in Python tests
+      run: |
+        pushd pennylane-lightning-kokkos
+        python3 -m pip install -r requirements.txt
+        BACKEND="OPENMP" python3 -m pip install .
+        popd
```

Comment:
Why is this necessary? Apparently it takes two minutes two install in the action, and we don't do the same for regular lightning

Reply:
It's required for tests checking the results of Catalyst against PL (`device=lightning.kokkos`) and it only takes 2min ;) We don't do the same for lightning as this is a requirement of the PL package and is installed via `pip install pennylane`.

Reply:
Hmm I mean I don't know that there is a reason the verify against lightning.kokkos specifically (in fact we probably shouldn't), but it would probably complicate the test functions 🤔

Reply:
The pytest `backend` option is currently referring to the entire backend devices used in both PL and Catalyst. We can update this to have a fix device for PL but it requires more work on going through all tests and ensure the `qml.device` separation between PL and Catalyst. The tests will look more hairy which I didn't like at that time.

Reply:
Can we not just pip install it then, rather than building the package from source?

Reply:
It's not available yet :(

Reply:
Regarding the slowdown, not all 2min about installing lightning.kokkos! Running tests using "lightning.kokkos" takes more than "lightning.qubit". We should start gaining from using "lightning.kokkos" for greater number of qubits than just a few.

running all tests locally,
"lightning.qubit"    : 0m 47.477s (real)
"lightning.kokkos" : 3m 1.463s (real)

Reply:
Wow that's quite the difference!

But I was specifically referring to the breakdown in the action:
![image](https://github.com/PennyLaneAI/catalyst/assets/29467667/0e7f83d6-d66a-4a91-8e44-8a217971822c)

Kokkos install is 2min and tests only 1min. I wonder why it takes 3 for you locally ?

Reply:
Hmm.. yes, yes! It depends on the number of OpenMP threads utilized by lightning and lightning.kokkos. I was testing devices serially (`OMP_NUM_THREADS=1`) while in CI this is set to 2.  I can get comparable results using  `OMP_NUM_THREADS=2`.